src/functions.php

```diff
@@ -188,7 +188,7 @@ function inspect_all($promises)
  * the promises were provided). An exception is thrown if any of the promises
  * are rejected.
  *
- * @param mixed $promises Iterable of PromiseInterface objects to wait on.
+ * @param PromiseInterface[] $promises Iterable of PromiseInterface objects to wait on.
```

Comment:
I am not sure about this.. wont PHPStan complain if I pass an iterator or an `ArrayAccess` object?

Reply:
Oh, yes. That's true. Should be `iterable<PromiseInterface>`.

Reply:
phpstorm does not support this phpdoc yet AFAIK. so it's tricky to use.

Reply:
Well, this is unacceptable as it is right now, since it's not an array. It was better as mixed!